src/components/ClusterOverview/ClusterOverview.js

```diff
@@ -15,70 +17,80 @@ import Capacity from './Capacity/Capacity';
 import Utilization from './Utilization/Utilization';
 import TopConsumers from './TopConsumers/TopConsumers';
 
-export const ClusterOverview = props => {
-  const mainComponent = (
-    <GridItem lg={6} md={12} sm={12}>
-      <Grid>
-        <GridItem span={6}>
-          <Health {...props.healthData} />
-        </GridItem>
-        <GridItem span={6}>
-          <Compliance {...props.complianceData} />
-        </GridItem>
-        <GridItem span={12}>
-          <Capacity {...props.capacityStats} />
-        </GridItem>
-        <GridItem span={12}>
-          <Utilization {...props.utilizationStats} />
-        </GridItem>
-      </Grid>
-    </GridItem>
-  );
+import { ClusterOverviewContext } from './ClusterOverviewContext';
 
-  return (
-    <Dashboard>
-      <DashboardHeader>Cluster Dashboard</DashboardHeader>
-      <DashboardBody>
-        <Grid>
-          <MediaQuery key="main-medium" maxWidth={MEDIA_QUERY_LG - MEDIA_QUERY_EXCLUSIVE_DEVIATION}>
-            {mainComponent}
-          </MediaQuery>
-          <GridItem key="left" lg={3} md={12} sm={12}>
-            <Grid>
-              <GridItem lg={12} md={6} sm={12}>
-                <Details {...props.detailsData} />
-              </GridItem>
-              <GridItem lg={12} md={6} sm={12}>
-                <Inventory {...props.inventoryData} />
-              </GridItem>
-            </Grid>
-          </GridItem>
-          <MediaQuery key="main-large" minWidth={MEDIA_QUERY_LG}>
-            {mainComponent}
-          </MediaQuery>
-          <GridItem key="right" lg={3} md={12} sm={12}>
-            <Grid>
-              <GridItem lg={12} md={6} sm={12}>
-                <Events {...props.eventsData} />
-              </GridItem>
-              <GridItem lg={12} md={6} sm={12}>
-                <TopConsumers {...props.consumersData} />
-              </GridItem>
-            </Grid>
-          </GridItem>
-        </Grid>
-      </DashboardBody>
-    </Dashboard>
-  );
+const ClusterOverviewContextConsumer = ({ Component, dataPath }) => (
+  <ClusterOverviewContext.Consumer>{value => <Component {...get(value, dataPath)} />}</ClusterOverviewContext.Consumer>
+);
+ClusterOverviewContextConsumer.propTypes = {
+  Component: PropTypes.func.isRequired,
+  dataPath: PropTypes.oneOfType([PropTypes.string, PropTypes.array]).isRequired,
 };
 
-ClusterOverview.propTypes = {
-  detailsData: PropTypes.object.isRequired,
-  inventoryData: PropTypes.object.isRequired,
-  healthData: PropTypes.object.isRequired,
-  complianceData: PropTypes.object.isRequired,
-  capacityStats: PropTypes.object.isRequired,
-  utilizationStats: PropTypes.object.isRequired,
-  eventsData: PropTypes.object.isRequired,
-  consumersData: PropTypes.object.isRequired,
-};
+const MainCards = () => (
+  <GridItem lg={6} md={12} sm={12}>
+    <Grid>
+      <GridItem span={6}>
+        <ClusterOverviewContextConsumer Component={Health} dataPath="healthData" />
```

Comment:
Wouldn't it be better to consume the data inside the component itself? There, the component can decide what data it needs (if any). This helps with sharing the resources (e.g. vms, pods)

Reply:
meaning: each card decides what it consumes

Reply:
+1 i would like to have such flexibility

Reply:
Good question.

Such approach would limit reusability of the cards (not sure if needed). It would state requirement on
- existence of the Context (means Provider)
- specific structure of the Context

Moving manipulation with the consumer one step up leads to centralization of manipulation with the data structure and leaves the components (means cards) independent.

There's no requirement to use the `ClusterOverviewCOntextConsumer` component for all children. It's just recent case for the mock data. I expect that some of the children will be created by using the `ClusterOverviewContext.Consumer` directly.

I can imagine that other dashboards will implement the Provider "on their own" so this will help - they will not need to meet certain structure of the context data.

Reply:
What about having both :-) : 
- keeping the card components using props
- and wrapping them with on-liners to provide them with the context-data

Reply:
I think the specific structure is already prescribed by the props and it is still needed to specify the dataPath. We can still export two components per card (one connected, one with props) if reusability issue arise.

Reply:
+1 you were a bit faster (:

Reply:
done